applications/luci-app-shadowsocks-libev/htdocs/luci-static/resources/view/shadowsocks-libev/rules.js

```diff
@@ -0,0 +1,119 @@
+				if (!uci.get_first(conf, 'ss_rules')) {
+					uci.set(conf, uci.add(conf, 'ss_rules', 'ss_rules'), 'disabled', '1');
+				}
+			})
+		]);
+	},
+	render: function(stats) {
+		var m, s, o;
+
+		m = new form.Map(conf, _('Redir Rules'),
+			_('On this page you can configure how traffics are to be \
+				forwarded to ss-redir instances. \
+				If enabled, packets will first have their src ip addresses checked \
+				against <em>Src ip/net bypass</em>, <em>Src ip/net forward</em>, \
+				<em>Src ip/net checkdst</em> and if none matches <em>Src default</em> \
+				will give the default action to be taken. \
+				If the prior check results in action <em>checkdst</em>, packets will continue \
+				to have their dst addresses checked.'));
+
+		s = m.section(form.NamedSection, 'ss_rules', 'ss_rules');
+		s.tab('general', _('General Settings'));
+		s.tab('src', _('Source Settings'));
+		s.tab('dst', _('Destination Settings'));
+
+		s.taboption('general', form.Flag, 'disabled', _('Disable'));
+		if (!stats[1]) {
+			ss.option_install_package(s, 'general');
+		}
+
+		o = s.taboption('general', form.ListValue, 'redir_tcp',
+			_('ss-redir for TCP'));
+		ss.values_redir(o, 'tcp');
+		o = s.taboption('general', form.ListValue, 'redir_udp',
+			_('ss-redir for UDP'));
+		ss.values_redir(o, 'udp');
+
+		o = s.taboption('general', form.ListValue, 'local_default',
+			_('Local-out default'),
+			_('Default action for locally generated TCP packets'));
+		ss.values_actions(o);
+		o = s.taboption('general', form.DynamicList, 'ifnames',
+			_('Ingress interfaces'),
+			_('Only apply rules on packets from these network interfaces'));
+		//ss.values_ifnames(o); // FIXME DeviceSelect with DynamicList?
+		s.taboption('general', form.Value, 'ipt_args',
+			_('Extra arguments'),
+			_('Passes additional arguments to iptables. Use with care!'));
+
+		src_dst_option(s, 'src', form.DynamicList, 'src_ips_bypass',
+			_('Src ip/net bypass'),
+			_('Bypass ss-redir for packets with src address in this list'));
+		src_dst_option(s, 'src', form.DynamicList, 'src_ips_forward',
+			_('Src ip/net forward'),
+			_('Forward through ss-redir for packets with src address in this list'));
+		src_dst_option(s, 'src', form.DynamicList, 'src_ips_checkdst',
+			_('Src ip/net checkdst'),
+			_('Continue to have dst address checked for packets with src address in this list'));
+		o = s.taboption('src', form.ListValue, 'src_default',
+			_('Src default'),
+			_('Default action for packets whose src address do not match any of the src ip/net list'));
+		ss.values_actions(o);
+
+		src_dst_option(s, 'dst', form.DynamicList, 'dst_ips_bypass',
+			_('Dst ip/net bypass'),
+			_('Bypass ss-redir for packets with dst address in this list'));
+		src_dst_option(s, 'dst', form.DynamicList, 'dst_ips_forward',
+			_('Dst ip/net forward'),
+			_('Forward through ss-redir for packets with dst address in this list'));
+
+		o = s.taboption('dst', form.Value, 'dst_ips_bypass_file', // FIXME FileBrowser
```

Comment:
It seems that `FileBrowser` has no effect, just like a normal `Value`, is it deprecated?
https://github.com/openwrt/luci/blob/176cbe12ea29006e4d4bf9dd3fe2ee708c010372/applications/luci-app-shadowsocks-libev/luasrc/model/cbi/shadowsocks-libev/rules.lua#L78-L85

Reply:
Use `form.FileUpload` as replacement.